qt/aqt/overview.py

```diff
@@ -63,6 +65,23 @@ def show(self) -> None:
     def refresh(self) -> None:
         self._refresh_needed = False
         self.mw.col.reset()
+
+        def fetch_data(
+            col: Collection,
+        ) -> Tuple[DeckDict, DeckId, bool]:
+            deck = col.decks.current()
+            deck_id = col.decks.selected()
+            have_buried = col.sched.have_buried()
+            return (deck, deck_id, have_buried)
+
+        QueryOp(
+            parent=self.mw,
+            op=fetch_data,
+            success=self._render,
+        ).run_in_background()
+
+    def _render(self, data: Tuple[DeckDict, DeckId, bool]) -> None:
+        self._current_deck, self._deck_id, self._have_buried = data
```

Comment:
Ideally we'd pass the fetched state into the functions that need it, instead of storing it as attributes on self. But if we changed the signature of functions like _renderBottom(), we'd break some add-ons that are overwriting it :-( So storing it as hidden properties seems reasonable.

I was about to point out the separate deck query in _renderPage(), but it looks like you already caught it. Had I noticed that at the start, I would have more strongly pushed for the do-the-bare-minimum approach, as a minor bugfix has turned into a bit of a refactor which we probably don't need right now :-) How about we put this PR on hold until 2.1.55 is out the door, and apply the following fix in the mean time?

```patch
diff --git a/qt/aqt/overview.py b/qt/aqt/overview.py
index 8d79c0034..9d64b0db6 100644
--- a/qt/aqt/overview.py
+++ b/qt/aqt/overview.py
@@ -277,7 +277,8 @@ class Overview:
         links = [
             ["O", "opts", tr.actions_options()],
         ]
-        if self.mw.col.decks.current()["dyn"]:
+        is_dyn = self.mw.col.decks.current()["dyn"]
+        if is_dyn:
             links.append(["R", "refresh", tr.actions_rebuild()])
             links.append(["E", "empty", tr.studying_empty()])
         else:
@@ -285,7 +286,8 @@ class Overview:
             # links.append(["F", "cram", _("Filter/Cram")])
         if self.mw.col.sched.have_buried():
             links.append(["U", "unbury", tr.studying_unbury()])
-        links.append(["", "description", tr.scheduling_description()])
+        if not is_dyn:
+            links.append(["", "description", tr.scheduling_description()])
         link_handler = gui_hooks.overview_will_render_bottom(
             self._linkHandler,
             links,
```

Regarding fetch-everything-at-once vs lots-of-little-queries, each background op has a slight performance overhead, and a code overhead as a function has to be split into fetch and do-after-fetch steps. That makes it more efficient to do everything in one go.

Reply:
(if there are still more mw.col calls that haven't been handled, another option is to just abandon this PR for now. This is part of the codebase we'll eventually want to move into a ts page, so it's not worth spending a lot of effort into improving it as it will be obsolete in the future)

Reply:
Yeah, I'll create a new PR with the small fix and abandon this one for now.

Reply:
Thanks Matthias, sorry for not being clearer from the start.

Reply:
No worries. I learned some stuff about QueryOp, so it wasn't wasted time :)